checker-qual/src/main/java/org/checkerframework/dataflow/qual/AssertMethod.java

```diff
@@ -0,0 +1,56 @@
+package org.checkerframework.dataflow.qual;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * {@code AssertMethod} is a method annotation that indicates that a method throws an exception if
+ * the value of a boolean argument is false. This can be used to annotate methods such as Junit's
+ * {@code Assertions.assertTrue(...)}.
+ *
+ * <p>The annotation enables flow-sensitive type refinement to be more precise. For example, after
+ *
+ * <pre>
+ *   Assertions.assertTrue(optional.isPresent());
+ * </pre>
+ *
+ * the Optional Checker can determine that {@code optional} has a value.
+ *
+ * <p>This annotation is a <em>trusted</em> annotation, meaning that it is not checked whether the
+ * annotated method really does throw an exception if the boolean expression is true.
+ *
+ * @checker_framework.manual #type-refinement Automatic type refinement (flow-sensitive type
+ *     qualifier inference)
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface AssertMethod {
+
+  /**
+   * The class of the exception thrown by this method. The default is {@link AssertionError}.
+   *
+   * @return class of the exception thrown by this method
+   */
+  Class<?> value() default AssertionError.class;
+
+  /**
+   * The one-based index of the boolean parameter that is tested. If the parameter is false, the
+   * method throws an exception. The check can be reversed by use of the {@link #exceptionalResult}
+   * element; if {@link #exceptionalResult} is true, then the method throws an exception if the
+   * parameter is true.
+   *
+   * @return the one-based index of the boolean parameter that is tested
+   */
+  int parameter() default 1;
+
+  /**
+   * On which value for {@link #parameter} does the method throw an exception?
+   *
+   * @return the result for {@link #parameter} on which the method throw an exception
+   */
+  boolean exceptionalResult() default false;
```

Comment:
Use of "result" here could be misleading because this is not about the annotated method's result.  Rather, it is about the *value* of the boolean expression whose index is passed as `parameter`.

More seriously, though, there are multiple names used for this concept in the source code:
```
exceptionalResult
exceptionOnResult
isAssertTrue
```

This suggests we have not yet found the best name (or we have not settled on it).  Let's find a good name and use it consistently in the documentation and in the code.  I lean toward `isAssertTrue` or `exceptionalValue` or `exceptionalBooleanValue`.

Reply:
Rename to `isAssertFalse = true`  defaults to `false`